codegen/templates/resource.go.tpl

```diff
@@ -0,0 +1,201 @@
+package {{ .DirName }}
+
+// DO NOT EDIT
+// This code is generated.
+
+import (
+	"fmt"
+	"log"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/hashicorp/vault/api"
+	"github.com/terraform-providers/terraform-provider-vault/util"
+)
+
+{{- if .SupportsWrite }}
+const {{ .LowerCaseDifferentiator }}Endpoint = "{{ .Endpoint }}"
+{{- else }}
+// This resource supports "{{ .Endpoint }}".
+{{ end }}
+
+func {{ .UpperCaseDifferentiator }}Resource() *schema.Resource {
+	fields := map[string]*schema.Schema{
+		"path": {
+			Type:        schema.TypeString,
+			Required:    true,
+			ForceNew:    true,
+			Description: "Path to backend to configure.",
+			StateFunc: func(v interface{}) string {
+				return strings.Trim(v.(string), "/")
+			},
+		},
+		{{- range .Parameters }}
+		"{{ .Name }}": {
+			{{- if (eq .Schema.Type "string") }}
+			Type:        schema.TypeString,
+			{{- end }}
+			{{- if (eq .Schema.Type "boolean") }}
+			Type:        schema.TypeBool,
+			{{- end }}
+			{{- if (eq .Schema.Type "integer") }}
+			Type:        schema.TypeInt,
+			{{- end }}
+			{{- if (eq .Schema.Type "array") }}
+			Type:        schema.TypeList,
+			Elem:        &schema.Schema{Type: schema.TypeString},
+			{{- end }}
+			{{- if .Required }}
+			Required:    true,
+			{{- else }}
+			Optional:    true,
+			{{- end }}
+			{{- if .Schema.DisplayAttrs.Sensitive }}
+			Sensitive:   true,
+			{{- end }}
+			Description: "{{ .Description }}",
+			{{- if .IsPathParam }}
+			ForceNew: true,
+			{{- end}}
+		},
+		{{- end }}
+	}
+	return &schema.Resource{
+		{{- if .SupportsWrite }}
+		Create: create{{ .UpperCaseDifferentiator }}Resource,
+		Update: update{{ .UpperCaseDifferentiator }}Resource,
+		{{- end }}
+		{{- if .SupportsRead }}
+		Read:   read{{ .UpperCaseDifferentiator }}Resource,
+		Exists: resource{{ .UpperCaseDifferentiator }}Exists,
+		{{- end }}
+		{{- if .SupportsDelete }}
+		Delete: delete{{ .UpperCaseDifferentiator }}Resource,
+		{{- end }}
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+		Schema: fields,
+	}
+}
+
+{{- if .SupportsWrite }}
+func create{{ .UpperCaseDifferentiator }}Resource(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*api.Client)
+	path := d.Get("path").(string)
+	vaultPath := util.ParsePath(path, nameEndpoint, d)
+	log.Printf("[DEBUG] Creating %q", vaultPath)
```

Comment:
Somebody somewhere along the way told me this was a requirement on Terraform's side. I can't find any documentation to the effect, nor can I remember which of the many PRs it was on. :-) I do note, though, that there are examples of it in their docs like [this one](https://www.terraform.io/docs/extend/writing-custom-providers.html). Moving away from it would be risky unless we can find a good reason and example.